### PR TITLE
Feat: Integrated Signup API

### PR DESCRIPTION
**Pull request by scjuly19**
- Added sign up feature with mock backend API
- fixed some errors on redirection and environment variables  occurring on netlify

### CODE REVIEW

File: src/context/dataContext/dataContext.js [hunk 1]
```diff
@@ -12,19 +12,29 @@ const DataProvider = ({ children }) => {
   const [state, dispatch] = useReducer(dataReducer, INITIAL_STATE);
   useEffect(() => {
     const fetchData = async () => {
-      dispatch({type:actionTypes.fetchData})
+      let didCancel = false;
+
+      dispatch({ type: actionTypes.fetchData });
       try {
```
Comment (shilpe26) [inline src/context/dataContext/dataContext.js:18]: Good use of try and catch👍🏻

File: src/pages/Login.js [hunk 2]
```diff
@@ -1,27 +1,21 @@
-import React ,{useState} from "react";
+import React, { useState } from "react";
 import "../styles/login.css";
```
Comment (shilpe26) [inline src/pages/Login.js:2]: make these two CSS links into one link of CSS, either importing any one of them in one file or making a separate CSS file to import all the CSS links into one file

File: src/pages/Signup.js [hunk 3]
```diff
@@ -2,38 +2,63 @@ import React from 'react'
 import {
   Link
 } from "react-router-dom";
+import { useAuthContext } from '../context/authContext/authContext';
+import { useAuthForm } from '../hooks/useAuthForm';
 export function Signup() {
+  const{signUp}=useAuthContext();
+  const { inputs, navigate,handleChange } = useAuthForm();
+
```
Comment (shilpe26) [inline src/pages/Signup.js:10]: remove extra space if not in use
👍🏻

**Review by shilpe26 — APPROVED** [review]
Update with the suggestions and you are good to merge👍🏻

**Review by shrey27 — APPROVED** [review]
Sign up feature is working fine.
Overall code looks good, but try not to hardcode endpoints and routes, instead save them in one file and pass them as variables. This makes it more clean and reusable.

PR is ready to merge

File: src/pages/Login.js (hunk 2 above)
Comment (scjuly19) [reply src/pages/Login.js:2]: Thanks for the sugestion. I have updated the PR with these changes.

**Pull request merged**